feat: update instruction with latest installation for helm (#333)
someengineering/resoto.com: docs/getting-started/install-resoto/kubernetes.md

pagination_next: getting-started/configure-cloud-provider-access/index
---

> # Install Resoto with Kubernetes(Helm Chart)

Reviewer: ```suggestion
# Install Resoto with Kubernetes
```

Author: I think install with Kubernetes is misleading as there are a lot of methods to install via helm, kubernetes manifest, terraform package. We should be more specific about what we are making users do.

Reviewer: It is the same with Docker; there are multiple ways to install. However, we only support one method.

Author: Yes, so we should highlight the way we support which is via the helm-chart. Not sure if there is any other way to run docker containers except via the docker agent. A container can be packaged differently to run on top of these orchestration platforms whether be kubernetes, swarm or compone.

Reviewer: I disagree. Please revert this change, but we can describe/clarify below that a Helm chart is provided/used.

Author: I am still not convinced as `With Kubernetes` sounds vague, we are talking about installing resoto on a Platform via a installation method(helm in this regards).

$ kubectl wait --for=condition=ready arangodeployment/single-server
```

> ### Configure Helm Values File

Author: We already have a default helm values file created which in turn is being modified by the user not being created from scratch.

> ### Configure Helm Values File

Reviewer: This guide is suggesting that users create a "barebones" `resoto-values.yaml` file, instead of modifying the `values.yaml`.

> ### Configure Helm Values File

Author: But is there a need for user to create the file, given the changes which we are telling the user to make in that are already present as defaults?

> ### Configure Helm Values File

Reviewer: They would need to fetch the file, in that case. Either way, it will be instructions to create the file, whether it be from the repository or from scratch. (We should add a `curl` command below if we want users to fetch a copy from the repo.)

> ### Configure Helm Values File

Author: They wouldn't need to fetch it as the defaults would be fetched via the `helm install` itself. If I install arango & just do `helm install resoto resoto/resoto --set image.tag=latest`, my resoto will point to default arango installation settings. Or am I missing some usecase over here 😅

> ### Configure Helm Values File

Reviewer: `helm install` is not executed until later. I believe the intention here was to configure the values file first and supply it on install.

> ### Configure Helm Values File

Author: Why I even need to fetch the values file when I can list all the helm defaults as well override everything via `--set` or providing a `<<EOF` during my installation. Ideally `Getting Started` for any project means to get the project up & running with the default and minimal configuration, and if we expect the user to create the values.yaml then we should have a detailed section about `How to configure Resoto with Kubernetes`

> ### Configure Helm Values File

Reviewer: I'm not disagreeing with you here, but your changes do not make sense unless we change the ordering/structure of this guide -- you ask users to edit a file that does not yet exist. 😉 Let's make the small changes to point to the new repo first, then we can discuss the best set of ordered instructions to get users up and running. I do think that it is much more user-friendly for users to edit an existing file than to create a new one.

> ### Configure Helm Values File

Author: Agreed, any amount of documentation is not more than enough for each and every user 😅

```yaml title="resoto-values.yaml"
resotocore:
  graphdb:
    server: http://single-server:8529
```

> By default in the values.yaml file, this is the minimum configuration, which points to an empty ArangoDB database with default username and password.

Reviewer: ```suggestion
This is the minimum configuration, which points to an empty ArangoDB database with default username and password.
```

Author: By default if you see the values.yaml, we point the server to the same URL https://github.com/someengineering/helm-charts/blob/1e6b3c3661923b59f3b86793fd906c74a5d254f4/charts/resoto/values.yaml#L89, so I dont know why are we wanting the user to make the changes which are already defaulted to the same.

Reviewer: This sentence and the above can be replaced with instructions to fetch the values file from the repository.

The installation will create a separate database and password and secure the database installation with a generated password. You can find the generated database password in the secret `arango-user`.

> See [`helm-chart`](https://github.com/someengineering/helm-charts/tree/main/charts/resoto) for a list of configurable values.

Reviewer: ```suggestion
See [`values.yaml`](https://github.com/someengineering/helm-charts/blob/main/charts/resoto/values.yaml) for a list of configurable values.
```

Reviewer: Hm, good point.

```suggestion
See the [`someengineering/helm-chart` GitHub repository](https://github.com/someengineering/helm-charts/tree/main/charts/resoto) for a list of configurable values.
```

### Install Helm Chart

> Add the resoto helm repository:

Reviewer: ```suggestion
Add the Resoto Helm chart repository:
```

```bash
$ helm repo add resoto https://helm.resoto.com
```

Next, install Resoto using Helm:

```bash

> $ helm install resoto resoto/resoto -f <values.yaml>

Reviewer: ```suggestion
$ helm install resoto resoto/resoto --set image.tag={{latestRelease}} -f resoto-values.yaml
```

Author: We already have listed image.tag as a configurable value & have highlighted it in the readme too, by default we should point it to a LTS release as most of the OSS projects do. For a user I firmly believe installing a project should be a one-touch setup.

Reviewer: Production setups should be pinning a Resoto version. Users can of course set this value to `latest` or omit it, but we should not condone this practice.

Also, `{{latestRelease}}` will be the latest stable release.

Author: > Also, {{latestRelease}} will be the latest stable release.

Yes but user will have to find that latestRelease by themselves right. We haven't templated it anywhere or have we 🤔

Reviewer: No, `{{latestRelease}}` is replaced with the tag.

Author: I am still not convinced with this change but I would leave it at your discretion if we expect the user to find the tag first & then replace it.

Reviewer: The user is not expected to find the tag and replace it; this becomes a copy-pastable command. Did you test or verify your changes in this PR?

Author: I had the confusion here but looks like this {{latestRelease}} is being rendered by docusaurus & being replaced with 2.4.0 in actual render

```

> And just like that, you have Resoto running in your Kubernetes cluster! A collect run will begin automatically. This first collect usually takes less than 3 minutes.

Reviewer: ```suggestion
And just like that, you have Resoto running in a Kubernetes cluster! A collect run will begin automatically. This first collect usually takes less than 3 minutes.
```

Author: Most of the language of article is directed to a user perspective, and last making it `your cluster` was more appropriate slang.

Reviewer: "Your" suggests that a user only has a single Kubernetes cluster, or that they themselves own it. "A" is more appropriate here.
